packages/falcon-client/src/serviceWorker/sw.js

```diff
@@ -39,10 +39,26 @@ if (CONFIG.precache) {
   precacheAndRoute(ENTRIES, {});
 }
 
+async function cleanResponse(response) {
+  const clonedResponse = response.clone();
+
+  // Not all browsers support the Response.body stream, so fall back to reading
+  // the entire body into memory as a blob.
+  const body = 'body' in clonedResponse ? clonedResponse.body : await clonedResponse.blob();
+
+  // new Response() is happy when passed either a stream or a Blob.
+  return new Response(body, {
+    headers: clonedResponse.headers,
+    status: clonedResponse.status,
+    statusText: clonedResponse.statusText
+  });
+}
+
 const router = new Router();
 self.addEventListener('fetch', event => {
-  const responsePromise = router.handleRequest(event);
+  let responsePromise = router.handleRequest(event);
   if (responsePromise) {
+    responsePromise = responsePromise.then(res => (res.redirected ? cleanResponse(res) : res));
     event.respondWith(responsePromise);
```

Comment:
can not we do 
```suggestion
     event.respondWith(responsePromise.then(res => (res.redirected ? cleanResponse(res) : res)));
```
?
then `responsePromise` can be still const.

The second thing is why we need to do "response to blob rendering" only when there is redirect? should not we do it for all possible responses?